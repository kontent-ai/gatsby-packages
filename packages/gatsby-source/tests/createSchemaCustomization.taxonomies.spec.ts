import { kontentTaxonomiesCreateSchemaCustomization } from '../src/createSchemaCustomization.taxonomies';
import { createMock } from 'ts-auto-mock';
import { Actions, CreateSchemaCustomizationArgs } from 'gatsby';
import { mocked } from 'ts-jest/dist/util/testing';

describe('createSchemaCustomization', () => {
  it('create fixed taxonomy definition', () => {
    const api = createMock<CreateSchemaCustomizationArgs>({
      actions: createMock<Actions>({
        createTypes: jest.fn(),
      }),
    });

    kontentTaxonomiesCreateSchemaCustomization(api);
    const createTypesMock = mocked(api.actions.createTypes, true);
    expect(createTypesMock).toBeCalledTimes(1);
    // Fix schema
    expect(createTypesMock.mock.calls[0][0]).toMatchSnapshot();
  });
});
