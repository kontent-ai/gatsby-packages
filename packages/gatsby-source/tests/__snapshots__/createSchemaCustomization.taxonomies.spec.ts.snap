// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSchemaCustomization create fixed taxonomy definition 1`] = `
"# All __UPPER_CASE__  wildcards will be replaced by the configuration

type kontent_taxonomy implements Node {
  id: ID!
  system: kontent_taxonomy_system!
  terms: [kontent_taxonomy_term]
}

type kontent_taxonomy_system {
  codename: String!
  id: String!
  last_modified: Date! @dateformat
  name: String!
}

type kontent_taxonomy_term {
  codename: String
  name: String
  terms: [kontent_taxonomy_term]
}
"
`;
